service/history/archival_queue_task_executor.go

```diff
@@ -244,13 +250,9 @@ func (e *archivalQueueTaskExecutor) addDeletionTask(ctx context.Context, logger
 		mutableState,
 		e.shardContext.GetConfig(),
 	)
-	closeTime, err := mutableState.GetWorkflowCloseTime(ctx)
-	if err != nil {
-		return err
-	}
 	err = taskGenerator.GenerateDeleteHistoryEventTask(*closeTime, true)
 	if err != nil {
-		return nil
+		return err
```

Comment:
This is the bug we're fixing. The rest of this is just adding additional test coverage.